src/gpu/vulkan/SDL_gpu_vulkan.c

```diff
@@ -12254,12 +12254,14 @@ static Uint8 VULKAN_INTERNAL_CreateLogicalDevice(
     deviceCreateInfo.ppEnabledExtensionNames = deviceExtensions;
 
     VkPhysicalDeviceFeatures2 featureList;
-    if (features->usesCustomVulkanOptions) {
+    int minor = VK_VERSION_MINOR(features->desiredApiVersion);
+
+    if (features->usesCustomVulkanOptions && minor > 0) {
         featureList.sType = VK_STRUCTURE_TYPE_PHYSICAL_DEVICE_FEATURES_2;
         featureList.features = features->desiredVulkan10DeviceFeatures;
-        featureList.pNext = &features->desiredVulkan11DeviceFeatures;
+        featureList.pNext = minor > 1 ? &features->desiredVulkan11DeviceFeatures : (void *)deviceCreateInfo.pNext;
```

Comment:
Why does this pass `deviceCreateInfo.pNext` instead of just NULL?

Reply:
The pNext pointers need to be either a pointer that specifically extends the structure or NULL. So in this case if we're not passing the feature extension pointer it should be NULL. I'm pretty sure the line for 1.3 is wrong as well.

Reply:
Done!